anta/tests/routing/bgp.py

```diff
@@ -447,3 +448,213 @@ def test(self) -> None:
 
         if failures:
             self.result.is_failure(f"Failures: {list(failures.values())}")
+
+
+class VerifyBGPPeerMPCaps(AntaTest):
+    """
+    Verifies the multiprotocol capabilities of a BGP peer in a specified VRF.
+    Expected results:
+        * success: The test will pass if the BGP peer's multiprotocol capabilities are advertised, received, and enabled in the specified VRF.
+        * failure: The test will fail if BGP peers are not found or multiprotocol capabilities are not advertised, received, and enabled in the specified VRF.
+    """
+
+    name = "VerifyBGPPeerMPCaps"
+    description = "Verifies the multiprotocol capabilities of a BGP peer in a specified VRF"
+    categories = ["routing", "bgp"]
+    commands = [AntaCommand(command="show bgp neighbors")]
+
+    class Input(AntaTest.Input):
+        """
+        Input parameters of the testcase.
+        """
+
+        bgp_peers: List[BgpPeers]
+        """List of BGP peers"""
+
+        class BgpPeers(BaseModel):
+            """
+            This class defines the details of a BGP peer.
+            """
+
+            peer: Union[IPv4Address, IPv6Address]
+            """IPv4/IPv6 BGP peer"""
+            vrf: str = "default"
+            """VRF context"""
```

Comment:
Can you add more unit tests including IPv6, multiple BGP peers in different VRFs, etc.

Reply:
You can set it up in the lab to make sure you have the proper outputs.